### ISSUE TITLE
tt::tt_metal::write_tensor doesn't properly support tiled destination tensors

### ISSUE THREAD

**Issue by jaykru-tt** [tenstorrent/tt-metal#22325]
### Describe the bug

For a tiled on-device destination tensor `B` with *tile padding*, calls of the form `tt::tt_metal::write_tensor(A,B)` where `A` is a tiled host tensor with the same logical shape as `B` fail due to what looks like a bad volume check (not specially handling padding) in `replicate_to_mesh_tensor`, comments my own:

```
// NOTE: This doesn't handle tiling correctly. The host_buffer is returned
// WITHOUT tile padding, so we get an obvious size mismatch. I think what
// I'll need to do is add the padding myself for the host buffer.
auto data_to_write = host_buffer::get_as<T>(storage.buffer);
const auto expected_packed_buffer_size_bytes = tensor_spec.compute_packed_buffer_size_bytes();
const auto input_size_bytes = data_to_write.size() * sizeof(T);
```

Indeed, padding out the host tensor with Tensor::pad_to_tile does the trick.

### Steps to reproduce the issue

I've added a failing test `TrivialTnnFixedTest.WriteTensorTest` in the branch  `origin/jkruer/tt-train_meshtrace`.

I actually wasn't able to reproduce this with the following Python snippet, so how we create host tensors with `ttml_create_owned_tensor` may be the underlying culprit:

```
import ttnn
import torch

device = ttnn.open_device(device_id=0)
shape = (20,20)
a = torch.ones(shape)
a = ttnn.from_torch(a, dtype=ttnn.bfloat16, layout=ttnn.TILE_LAYOUT)
a_dev = a.to(device)
ttnn.copy_host_to_device_tensor(a, a_dev)
ttnn.close_device(device
```

**Comment by dmakoviichuk-tt** [comment]
Closing it for now.
